tower-http/src/metrics/traffic/service.rs

```diff
@@ -0,0 +1,76 @@
+use super::{Callbacks, ResponseBody, ResponseFuture, TrafficLayer};
+use crate::classify::MakeClassifier;
+use http::{Request, Response};
+use http_body::Body;
+use std::{
+    fmt,
+    task::{Context, Poll},
+    time::Instant,
+};
+use tower_service::Service;
+
+/// Middleware for adding high level traffic metrics to a [`Service`].
+///
+/// See the [module docs](crate::metrics::traffic) for more details.
+///
+/// [`Layer`]: tower_layer::Layer
+/// [`Service`]: tower_service::Service
+#[derive(Debug, Clone)]
+pub struct Traffic<S, M, Callbacks> {
+    pub(super) inner: S,
+    pub(super) make_classifier: M,
+    pub(super) callbacks: Callbacks,
+}
+
+impl<S, M, Callbacks> Traffic<S, M, Callbacks> {
```

Comment:
`Callbacks` should be `CallbacksT` here, seemingly?  Just to follow the same pattern as the actual service impl.

Reply:
Another question, too: maybe just use `C` for all generic type parameters where we're taking a `Callbacks` implementation? It feels like a code smell to have so many spots where we need to differentiate with `CallbacksT` in order to avoid confusion.

Reply:
The pattern I've been following elsewhere in tower-http is to fully spell out the name of the generic type if there is another type close by with a similar name. For example `C` might be `Classifier` or `Callbacks`.

And then since you cannot do `Callbacks: Callbacks` in bounds I would use `CallbacksT` for the type in those places.

Not a perfect pattern but thats at least the reasoning.

Reply:
Yeah, I mean, it makes sense.  I personally find the fully-spelled out versions a little harder to remember vs the acronyms, but I know it's an established pattern in Tower so there's no eliminating it altogether. :)

My gut feeling is that it should just be consistently applied, and you're using `CallbacksT` in some places, so that seems like the pattern that should be followed, IMO.